### ISSUE TITLE
Sega Saturn keyboard support

### ISSUE THREAD

**Issue by sikthehedgehog** [hsoft/collapseos#12]
Seeing as this OS can work on a Master System, it's worth considering. We already got it working on a Mega Drive (plug shape is different but signals are the same, so it only needs a passive adapter or just outright replacing the plug) and there's already at least one game supporting it. This same setup should also work on a Master System (it requires TH and TR to be outputs so it won't work on a Mark III, but it's probably still worth a shot).

This probably will require changing `kbd.asm` to not assume it's a PS/2 keyboard (probably easier to rename the file, repurpose that code as a PS/2 driver and then write a wrapper around it in a new `kbd.asm`).

This information should be relevant (albeit was written with Mega Drive in mind):
* Keyboard: https://plutiedev.com/saturn-keyboard
* Plug pinout: https://plutiedev.com/io-pinout

Addenum:
* Bit 6 = TH
* Bit 5 = TR
* Bit 4 = TL
* Bits 3-0 = DATA3-0

PS: probably anything that can be supported with TL and DATA3-0 as inputs should work on a Western Master System (that includes a large amount of peripherals actually).

**Comment by sikthehedgehog** [comment]
Detail I forgot to mention, but there's a PS/2 to Saturn keyboard adapter too (and in fact it's more common and much cheaper than the actual standalone keyboard), of course that one will work too (same protocol as far as the console is aware).

**Comment by hsoft** [comment]
Yes, very good suggestion. This suggestion was already made on the SMS Power forums ( http://www.smspower.org/forums/17506-OperatingSystemForSMS ) and I thought it was a very good idea. It still is :)
